velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -3007,3 +3008,50 @@ TEST_F(DateTimeFunctionsTest, dateFunctionTimestampWithTimezone) {
       dateFunction(
           (-18297 * kSecondsInDay + 6 * 3'600) * 1'000, "America/Los_Angeles"));
 }
+
+TEST_F(DateTimeFunctionsTest, currentDate) {
+  // Since the execution of the code is slightly delayed, it is difficult for us
+  // to get the correct value of current_date. If you compare directly based on
+  // the current time, you may get wrong result at the last second of the day,
+  // and current_date may be the next day of the comparison value. In order to
+  // avoid this situation, we compute a new comparison value after the execution
+  // of current_date, so that the result of current_date is either consistent
+  // with the first comparison value or the second comparison value, and the
+  // difference between the two comparison values is at most one day.
+
+  auto emptyRowVector = makeRowVector(ROW({}), 1);
+
+  // First, Do not set the timezone, so the timezone obtained from QueryConfig
+  // will be nullptr.
+  auto tp0 = std::chrono::system_clock::now();
```

Comment:
variable names a1, a2, a3 are anti-pattern; please, rename; it may be helpful to split this test into 2: with and without timezone specified.